packages/docusaurus-theme-classic/src/theme/Tabs/index.tsx

```diff
@@ -48,6 +49,25 @@ function Tabs(props: Props): JSX.Element {
   const {tabGroupChoices, setTabGroupChoices} = useUserPreferencesContext();
   const [selectedValue, setSelectedValue] = useState(defaultValue);
   const tabRefs: (HTMLLIElement | null)[] = [];
+  const {enableScrollMonitor, disableScrollMonitor} = useScrollMonitorContext();
+  const {measureTop, restoreTop} = useRestoreTop();
+
+  useLayoutEffect(() => {
+    // prevent `restoreTop` from triggering the navbar auto hide/show by
+    // disabling the scroll monitoring temporarily
+    disableScrollMonitor();
+    restoreTop();
+    setTimeout(() => {
+      enableScrollMonitor();
+    }, 150);
```

Comment:
magic value, can we get rid of it?

Reply:
what do we have to wait for? can't the scrollMonitor be re-enabled immediately?

Is it to avoid `useScrollPosition` calling its callback?

Reply:
yeah the entire workflow here is very hacky, but it goes like so:

- when a 'tab group' is attempted to change (i.e. clicking on a tab), the 'top' of the element is measured (distance from top relative to the viewport)
- when the component re-renders and the `tabGroupChoices` state has changed, the above `useLayoutEffect` is triggered
- 'scroll monitoring' is marked as disabled (i.e. whether the `handleScroll` callback in `useScrollPosition` will run)
- `restoreTop` is called to scroll the window such that the 'top' of the element is back to wherever it was at the point it got measured. This gives the *appearance* that the page has not shifted, relative to the tab that was clicked
- the `window.scroll` call within `restoreTop` triggers the `handleScroll` callback via the `scroll` event listener. However, this does *not* occur synchronously, it goes onto the next event loop (at the very least?)
- the goal at this point is to re-enable scroll monitoring only *after* the `handleScroll` has been called, so the `setTimeout` is used to defer it a bit. If `enableScrollMonitor` is just called immediately, it will be called *before* `handleScroll` triggers

Realistically I believe a setTimeout value of `0` is enough just to call it on the next event loop, but I just used 150 as an arbitrary 'just in case' magic number.

I'm not a big fan of the hacky solution in general, but I don't immediately see another clean way to achieve:
- mess with the scroll position behind the scenes to 'appear' as though the page didn't shift, relative to the tab
- yet *don't* trigger any `scroll` events that would be triggered by this particular `window.scroll` usage

I'm happy to change the timeout value to 0 instead and add an explanatory comment, but if you have some better ideas for the implementation I'm happy to hear that